channeldb/graph.go

```diff
@@ -697,7 +697,7 @@ func (c *ChannelGraph) PruneGraph(spentOutputs []*wire.OutPoint,
 
 	var chansClosed []*ChannelEdgeInfo
 
-	err := c.db.Update(func(tx *bbolt.Tx) error {
+	err := c.db.Batch(func(tx *bbolt.Tx) error {
```

Comment:
`chansClosed` needs to be nilled during each invocation

Reply:
Nice catch! Fixed here and below, reminds me to do a quick sweep to see if we have any other possible bugs here due to the potential of batch commit retries.